openide.awt/test/unit/src/org/netbeans/modules/openide/awt/StatefulActionProcessorTest.java

```diff
@@ -0,0 +1,1386 @@
+            "    public static class B implements ActionListener {\n" +
+            "       public B(ActionModel mdl) {} \n" +
+            "      public void actionPerformed(ActionEvent e) {}\n" +
+            "    }\n" +
+            "}\n"
+        );
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        boolean r = AnnotationProcessorTestUtils.runJavac(getWorkDir(), null, getWorkDir(), null, os);
+        if (!os.toString().contains("Method addChangeListener not found")) {
+            fail("Missing add listener must be reported" + os);
+        }
+    }
+
+    /**
+     * issues an error if the specified trigger method does not exist
+     */
+    public void testMissingListenerMethod() throws Exception {
+        clearWorkDir();
+        AnnotationProcessorTestUtils.makeSource(getWorkDir(), "test.A", 
+            "import org.openide.awt.ActionRegistration;\n" +
+            "import org.openide.awt.ActionReference;\n" +
+            "import org.openide.awt.ActionState;\n" +
+            "import org.openide.awt.ActionID;\n" +
+            "import org.openide.util.actions.Presenter;\n" +
+            "import java.awt.event.*;\n" +
+            "import javax.swing.event.*;\n" +
+            "import java.util.List;\n" +
+            "import javax.swing.*;\n" +
+            "import org.netbeans.modules.openide.awt.StatefulActionProcessorTest.*;\n" +
+                    
+            "public class A {\n" +
+            "    @ActionID(category=\"Tools\",id=\"test.action\")" +
+            "    @ActionRegistration(displayName=\"AAA\", key=\"K\") " +
+            "    @ActionReference(path=\"manka\", position=11)" +
+            "    @ActionState(type = ActionModel5.class, property=\"prop\", "
+                    + "listenOn=CustomListener.class, listenOnMethod=\"bubu\")" +
+            "    public static class B implements ActionListener {\n" +
+            "       public B(ActionModel mdl) {} \n" +
+            "      public void actionPerformed(ActionEvent e) {}\n" +
+            "    }\n" +
+            "}\n"
+        );
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        boolean r = AnnotationProcessorTestUtils.runJavac(getWorkDir(), null, getWorkDir(), null, os);
+        if (!os.toString().contains("does not contain method bubu")) {
+            fail("Missing listener method must be reported" + os);
+        }
+    }
+    
+    @ActionID(id = "test.DefEnableAction", category="Foo")
+    @ActionRegistration(displayName = "TestAction", 
+            enabledOn = @ActionState()
+    )
+    public static class DefEnableAction implements ActionListener {
+
+        public DefEnableAction(DefaultActionModel model) {
+            instance2 = this;
+        }
+
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            received = e;
+        }
+    }
+    
+    /**
+     * Checks that the "enabled" property is used by default if property is not specified
```

Comment:
Is that really needed? The `enabledOn = @ActionState()` code looks magical...

Reply:
Eh ... this is rather a side effect of defaulting the guard property to `enabled`. It would look less scary as element annotation, I think.